<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CertificateManagementDialog</class>
 <widget class="QDialog" name="CertificateManagementDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>731</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Certificate Manager</string>
  </property>
  <property name="windowIcon">
   <iconset theme="certificate">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="rightMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QTreeView" name="certificates">
         <property name="acceptDrops">
          <bool>true</bool>
         </property>
         <property name="dragEnabled">
          <bool>true</bool>
         </property>
         <property name="dragDropMode">
          <enum>QAbstractItemView::DragDrop</enum>
         </property>
         <property name="defaultDropAction">
          <enum>Qt::MoveAction</enum>
         </property>
         <attribute name="headerVisible">
          <bool>false</bool>
         </attribute>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QToolButton" name="create_cert_button">
           <property name="toolTip">
            <string>Create new certificate</string>
           </property>
           <property name="text">
            <string>Create...</string>
           </property>
           <property name="icon">
            <iconset theme="document-new"/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="import_cert_button">
           <property name="toolTip">
            <string>Import certificate</string>
           </property>
           <property name="text">
            <string>Import...</string>
           </property>
           <property name="icon">
            <iconset theme="document-import"/>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QToolButton" name="export_cert_button">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Export certificate</string>
           </property>
           <property name="text">
            <string>Export...</string>
           </property>
           <property name="icon">
            <iconset theme="document-export"/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="delete_cert_button">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Delete certificate</string>
           </property>
           <property name="text">
            <string>Delete</string>
           </property>
           <property name="icon">
            <iconset theme="edit-delete"/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="title">
        <string>Certificate</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Display Name</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="cert_display_name"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Common Name</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="cert_common_name">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Expiration Date</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDateTimeEdit" name="cert_expiration_date">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Expires in</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="cert_livetime">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>??? days</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Fingerprint</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Notes</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QPlainTextEdit" name="cert_fingerprint">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QPlainTextEdit" name="cert_notes"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Host Filter</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="cert_host_filter">
            <property name="placeholderText">
             <string>gemini://*</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QCheckBox" name="cert_auto_enable">
            <property name="toolTip">
             <string>If this is checked, Kristall will automatically enable this certificate when visiting a URL matching the host filter</string>
            </property>
            <property name="text">
             <string>Auto-Enable Certificate</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>certificates</tabstop>
  <tabstop>create_cert_button</tabstop>
  <tabstop>import_cert_button</tabstop>
  <tabstop>export_cert_button</tabstop>
  <tabstop>delete_cert_button</tabstop>
  <tabstop>cert_display_name</tabstop>
  <tabstop>cert_common_name</tabstop>
  <tabstop>cert_expiration_date</tabstop>
  <tabstop>cert_host_filter</tabstop>
  <tabstop>cert_auto_enable</tabstop>
  <tabstop>cert_fingerprint</tabstop>
  <tabstop>cert_notes</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>CertificateManagementDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>CertificateManagementDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
